File: public/public-persistence-jooq/src/main/java/ch/difty/scipamato/publ/persistence/codeclass/JooqCodeClassRepo.java
package ch.difty.scipamato.publ.persistence.codeclass;

import static ch.difty.scipamato.common.persistence.TranslationUtilsKt.trimLanguageCode;
import static ch.difty.scipamato.publ.db.tables.CodeClass.CODE_CLASS;

import java.util.List;

import org.jetbrains.annotations.NotNull;
import org.jooq.DSLContext;
import org.springframework.cache.annotation.CacheConfig;
import org.springframework.cache.annotation.Cacheable;
import org.springframework.stereotype.Repository;

import ch.difty.scipamato.publ.entity.CodeClass;

@Repository
@CacheConfig(cacheNames = "codeClasses")
public class JooqCodeClassRepo implements CodeClassRepository {

    private final DSLContext dslContext;

    public JooqCodeClassRepo(@NotNull final DSLContext dslContext) {
        this.dslContext = dslContext;
    }

    @NotNull
    @Override
    @Cacheable
    public List<CodeClass> find(@NotNull final String languageCode) {
        final String lang = trimLanguageCode(languageCode);
        // skipping the audit fields
        return dslContext
            .select(CODE_CLASS.CODE_CLASS_ID, CODE_CLASS.LANG_CODE, CODE_CLASS.NAME, CODE_CLASS.DESCRIPTION)
            .from(CODE_CLASS)
            .where(CODE_CLASS.LANG_CODE.eq(lang))
            .orderBy(CODE_CLASS.CODE_CLASS_ID)
            .fetchInto(CodeClass.class);
    }
}

File: public/public-persistence-jooq/src/main/kotlin/ch/difty/scipamato/publ/persistence/codeclass/JooqCodeClassRepo.kt
package ch.difty.scipamato.publ.persistence.codeclass

import ch.difty.scipamato.common.persistence.trimLanguageCode
import ch.difty.scipamato.publ.entity.CodeClass
import org.jooq.DSLContext
import org.springframework.cache.annotation.CacheConfig
import org.springframework.cache.annotation.Cacheable
import org.springframework.stereotype.Repository

@Repository
@CacheConfig(cacheNames = ["codeClasses"])
open class JooqCodeClassRepo(private val dslContext: DSLContext) : CodeClassRepository {

    @Suppress("MagicNumber")
    @Cacheable
    override fun find(languageCode: String): List<CodeClass> = dslContext
        .select(
            ch.difty.scipamato.publ.db.tables.CodeClass.CODE_CLASS.CODE_CLASS_ID,
            ch.difty.scipamato.publ.db.tables.CodeClass.CODE_CLASS.LANG_CODE,
            ch.difty.scipamato.publ.db.tables.CodeClass.CODE_CLASS.NAME,
            ch.difty.scipamato.publ.db.tables.CodeClass.CODE_CLASS.DESCRIPTION,
        )
        .from(ch.difty.scipamato.publ.db.tables.CodeClass.CODE_CLASS)
        .where(ch.difty.scipamato.publ.db.tables.CodeClass.CODE_CLASS.LANG_CODE.eq(trimLanguageCode(languageCode)))
        .orderBy(ch.difty.scipamato.publ.db.tables.CodeClass.CODE_CLASS.CODE_CLASS_ID)
        .map { r -> CodeClass(r[0] as Int?, r[1] as String, r[2] as String?, r[3] as String) }
}
